eth/hooks/engine_v2_hooks.go

```diff
@@ -361,24 +361,45 @@ func AttachConsensusV2Hooks(adaptor *XDPoS.XDPoS, bc *core.BlockChain, chainConf
 				rewardsMap[rwt.key] = rewardResults
 			}
 			// record the total reward into state db
-			totalMinted := state.GetTotalMinted(stateBlock).Big()
-			lastEpochNum := state.GetLastEpochNum(stateBlock)
-			if lastEpochNum.IsZero() {
-				// if `lastEpochNum` is zero, the total minted has not included tokens before TIPUpgradeReward
-				// calculate the tokens before TIPUpgradeReward and set to totalMinted
-				// for now no-do
+			totalMinted := big.NewInt(0)
+			totalBurned := big.NewInt(0)
+
+			nonce := stateBlock.GetNonce(common.MintedRecordAddressBinary)
+			if nonce == 0 {
+				// initialize MintedRecordAddress
+				state.PutMintedRecordOnsetEpoch(stateBlock, common.Uint64ToHash(epochNum))
+				state.PutMintedRecordOnsetBlock(stateBlock, common.Uint64ToHash(number))
+			} else {
+				epochNumIter := epochNum
+				for epochNumIter > 0 {
+					totalMinted = state.GetPostTotalMinted(stateBlock, epochNumIter-1).Big()
+					totalBurned = state.GetPostTotalBurned(stateBlock, epochNumIter-1).Big()
+					if totalMinted.BitLen() != 0 || totalBurned.BitLen() != 0 {
+						// if previous epoch has non-zero total minted or non-zero total burned, break the loop
+						break
+					}
+					epochNumIter--
+				}
 			}
 			totalMinted.Add(totalMinted, rewardSum)
 			bigPower256 := new(big.Int).Lsh(big.NewInt(1), 256)
 			bigMaxU256 := new(big.Int).Sub(bigPower256, big.NewInt(1))
 			// if overflow, set to maxU256 and log a warning
-			if totalMinted.Cmp(bigMaxU256) >= 0 {
+			if totalMinted.Cmp(bigMaxU256) > 0 {
 				totalMinted.Set(bigMaxU256)
```

Comment:
`totalMinted` will be greater than `bigMaxU256` one day.

Reply:
I think that will be very long time in the future. this PR can skip that.